src/vm/tests/contracts.rs

```diff
@@ -262,81 +262,53 @@ fn test_simple_naming_system() {
 
     env.sender = Some(p2.clone());
 
-    assert_eq!(
+    assert!(!is_committed(&
         env.execute_contract("names", "preorder",
-                             &symbols_from_values(vec![name_hash_expensive_0.clone(), Value::Int(1000)])).unwrap(),
-        Value::Bool(false));
+                             &symbols_from_values(vec![name_hash_expensive_0.clone(), Value::Int(1000)])).unwrap()));
 
     env.sender = Some(p1.clone());
-    assert_eq!(
+    assert!(is_committed(&
         env.execute_contract("names", "preorder",
-                             &symbols_from_values(vec![name_hash_expensive_0.clone(), Value::Int(1000)])).unwrap(),
-        Value::Bool(true));
-    assert_eq!(
+                             &symbols_from_values(vec![name_hash_expensive_0.clone(), Value::Int(1000)])).unwrap()));
+    assert!(!is_committed(&
         env.execute_contract("names", "preorder",
-                             &symbols_from_values(vec![name_hash_expensive_0.clone(), Value::Int(1000)])).unwrap(),
-        Value::Bool(false));
+                             &symbols_from_values(vec![name_hash_expensive_0.clone(), Value::Int(1000)])).unwrap()));
 
     // shouldn't be able to register a name you didn't preorder!
     env.sender = Some(p2.clone());
-    assert_eq!(
+    assert!(!is_committed(&
         env.execute_contract("names", "register",
-                             &symbols_from_values(vec![p2.clone(), Value::Int(1) , Value::Int(0)])).unwrap(),
-        Value::Bool(false));
+                             &symbols_from_values(vec![p2.clone(), Value::Int(1) , Value::Int(0)])).unwrap()));
+
     // should work!
     env.sender = Some(p1.clone());
-    assert_eq!(
+    assert!(is_committed(&
         env.execute_contract("names", "register",
-                             &symbols_from_values(vec![p2.clone(), Value::Int(1) , Value::Int(0)])).unwrap(),
-        Value::Bool(true));
+                             &symbols_from_values(vec![p2.clone(), Value::Int(1) , Value::Int(0)])).unwrap()));
 
 
     // try to underpay!
     env.sender = Some(p2.clone());
-    assert_eq!(
+    assert!(is_committed(&
         env.execute_contract("names", "preorder",
-                             &symbols_from_values(vec![name_hash_expensive_1.clone(), Value::Int(100)])).unwrap(),
-        Value::Bool(true));
-    assert_eq!(
+                             &symbols_from_values(vec![name_hash_expensive_1.clone(), Value::Int(100)])).unwrap()));
+    assert!(!is_committed(&
         env.execute_contract("names", "register",
-                             &symbols_from_values(vec![p2.clone(), Value::Int(2) , Value::Int(0)])).unwrap(),
-        Value::Bool(false));
+                             &symbols_from_values(vec![p2.clone(), Value::Int(2) , Value::Int(0)])).unwrap()));
```

Comment:
Is there a way to check the `(err` result from this transaction?  Also, is it possible to add other failure tests, like:

* trying to register the same name twice?
* trying to preorder the same name twice?
* trying to register a different name than was preordered?

Reply:
Not yet! There's no function in this PR for unpacking an error result. I'm looking for suggestions on how that function would look.

Reply:
Could this just be `(expects-err!` ?

Reply:
Yep, I'll add that.

Will leave it as an open question on whether or not we want a different set of functions that are like `expects` except that instead of performing a shallow return, throw a runtime error, causing all  execution of the tx to abort.

Reply:
Wait, I thought that's what `(expects!` does?  Doesn't it cause the current function to return the `throw value` immediately?

Reply:
Ah I see.  We could have something like `(unwrap!` or `(unwrap-err!` that would cause the entire tx to abort if the result was not `(ok` or `(err`, respectively.  This is distinct from `(expects!` in that `(expects!` is more like the `?` operator in Rust, as opposed to a naked `unwrap()`.

Reply:
Yep, exactly

Reply:
Cool, looks good!